Added Dockerfile and docker-compose

ba-tno opened the pull request:
A very blunt Dockerfile and docker-compose to run this on a system without node.
I'm cloning the repository into the image instead of using `COPY` or `ADD` to prevent cross-platform line-endings hassle. There's probably a more elegant way.

magjac commented:
Thanks for the contribution and apologies for the delay. I'm willing to merge this if you can add why and how to use it to the README.

ba-tno commented:
I've added a section in the README on how to use this. Regarding the why: installing make on windows appeared to be a hassle, and I already had docker available. Hope it's helpful to future users!

magjac commented on docker-compose.yml:
```diff
@@ -0,0 +1,10 @@
+    container_name: graphviz-editor
+    ports:
+      - "5000:5000"
+    restart: always
```
nit: missing newline here

magjac commented on docker/Dockerfile:
```diff
@@ -0,0 +1,18 @@
+RUN npm run build
+RUN npm install -g serve
+
+CMD ["serve", "-s", "build"]
```
nit:missing newline here

magjac commented on README.md:
```diff
@@ -27,6 +27,17 @@ npm run build
 Learn more from the Create React App [README](https://github.com/facebook/create-react-app#npm-run-build-or-yarn-build) and [User Guide](https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/template/README.md#deployment).
 
+## Docker ##
+To run this on a system without `make` and `node` you can use Docker.  
```
nit: trailing whitespace here

magjac commented on docker/Dockerfile:
```diff
@@ -1,18 +1,33 @@
-RUN apt install git -y
-RUN apt install make -y
+# Install git and make
+RUN apk update 
```
nit: trailing whitespace here

magjac reviewed (changes requested):
Can you please rebase this on the current master and get rid of the surplus commits and the merge commit?

ba-tno commented:
After a few rides on the git reset/rebase/force push merry go round I think I got rid of the extra commits and rebased it onto the `magjac:master` branch (I've set your master branch as upstream, created a local branch of it and rebased my docker branch onto that).
